examples/example.c

```diff
@@ -403,8 +418,14 @@ main(int argc, char **argv)
         if (has_arg(argc, argv, "unsample-tx")) {
             sentry_transaction_context_set_sampled(tx_ctx, 0);
         }
+        sentry_value_t custom_sampling_ctx = sentry_value_new_object();
+        sentry_value_set_by_key(
+            custom_sampling_ctx, "a", sentry_value_new_string("first_value"));
+        sentry_value_set_by_key(
+            custom_sampling_ctx, "b", sentry_value_new_int32(42));
         sentry_transaction_t *tx
-            = sentry_transaction_start(tx_ctx, sentry_value_new_null());
+            = sentry_transaction_start(tx_ctx, custom_sampling_ctx);
+        sentry_value_decref(custom_sampling_ctx);
```

Comment:
This needs to be documented properly, since it is necessary to `decref` to avoid leaks. We alternatively could `decref` inside the transaction start, just like we do with the `tx_ctx`, but users might want to reuse the one custom context multiple times.

I'm in favor of the first option, providing a clear example in the docs of this necessary `decref` cleanup

Reply:
I think there are two opposing arguments:

* A custom sampling context should exist close to where this "context" is. Thus, it could make sense to pass ownership on to us and let users create a new custom sampling context wherever required.
* A custom sampling context should be created only once and reused throughout the code. This requires client ownership, but the upside is fewer memory (de-)allocations.

With a fresh view of the current flow, the better model is to let us take ownership. This approach doesn't require the client to track usage, and they won't need to maintain access to their context references throughout their code. If code duplication of the context construction is a concern, nothing stops them from extracting that bit into a function.